packages/components/molecules/f-tabs/src/components/Tabs.vue

```diff
@@ -1,46 +1,147 @@
+        animate: {
+            type: Boolean,
+            default: false
         }
     },
-    data () {
-        const locale = globalisationServices.getLocale(tenantConfigs, this.locale, this.$i18n);
-        const localeConfig = tenantConfigs[locale];
-
+    data: () => ({
+        direction: null,
+        activeTab: '',
+        tabs: []
+    }),
+    provide () {
+        const component = this;
+        const tabsComponent = {};
+        Object.defineProperty(tabsComponent, 'activeTab', {
+            enumerable: true,
+            get: () => this.activeTab
+        });
+        Object.defineProperty(tabsComponent, 'animationDirection', {
+            enumerable: true,
+            get: () => this.direction
+        });
+        Object.defineProperty(tabsComponent, 'animate', {
+            enumerable: true,
+            get: () => this.animate
+        });
         return {
-            copy: { ...localeConfig }
+            register (tab) {
+                component.addTab(tab);
+            },
+            tabsComponent
         };
+    },
+    methods: {
+        /**
+         * This function is used to select the tab index and set the active tab while applying the correct animation
+         * uses the tab name to select index
+         * @param name
+         */
+        selectTabIndex (name) {
+            const previousIndex = this.tabs.findIndex(t => t.name === this.activeTab);
+            const newIndex = this.tabs.findIndex(t => t.name === name);
+            if (newIndex > previousIndex) {
+                this.direction = DIRECTION.RIGHT;
+            } else {
+                this.direction = DIRECTION.LEFT;
+            }
+            this.activeTab = name;
+        },
+        /**
+         * This function adds a tab that is calling the register tab callback
+         * @param tab
+         */
+        addTab (tab) {
+            if (tab.selected) {
+                this.activeTab = tab.name;
+            }
+            this.tabs = [...this.tabs, tab];
+        }
     }
 };
 </script>
 
 <style lang="scss" module>
 
+$tabs-link-colour         : $grey--darkest;
+$tabs-link-font-weight    : $font-weight-bold;
+$tabs-link-border-colour  : $brand--orange;
+
 .c-tabs {
+    width: 100%;
+    height: 100%;
+}
+.c-tabs-header {
+    width: 100%;
+    height: 42px;
+}
+.c-tabs-buttons {
     display: flex;
-    justify-content: center;
-    min-height: 80vh;
-    width: 80vw;
-    margin: auto;
-    border: 1px solid $red;
-    font-family: $font-family-base;
-    @include font-size(heading-m);
+    align-items: center;
+    list-style-type: none;
+    margin: 0;
+    padding: 0;
+    height: 100%;
+
+    @include media('<narrowMid') {
+        width: 100%;
+    }
+
+}
+.c-tabs-button {
+    padding: spacing(x1.5) spacing(x2);
+    cursor: pointer;
+    color: $tabs-link-colour;
+    border: none;
+    background-color: transparent;
+    outline: none;
+
+    @include media('<narrowMid') {
+        /* stylelint-disable */
```

Comment:
Not a big one by any means (can be in a future PR), but would be cleaner to use:

`/* stylelint-disable-next-line whatever-the-stylelint-rule-is */`

Reply:
@ashleynolan thanks for that, was trying to figure out what the one line version was, will make change in future PR